### PR TITLE
Ready: Upgrade Testing Libs & fix non-deterministic test.

### PR DESCRIPTION
**Pull request by alexmoore**
Upgrade Junit / Powermock / Mockito to latest versions, fix breaking changes.

Fix a non-deterministic test that @javajolt was seeing here: https://github.com/basho/riak-java-client/pull/582#issuecomment-164604357

### CODE REVIEW

File: src/test/java/com/basho/riak/client/core/util/DefaultCharsetTest.java [hunk 1]
```diff
@@ -31,7 +34,7 @@ public void tearDown()
     }
 
     @Test
-    public void testSystemDefaultCharsetIsReturnedIfNoPropertyIsDefined()
+    public void a_testSystemDefaultCharsetIsReturnedIfNoPropertyIsDefined()
```
Comment (hazen) [inline src/test/java/com/basho/riak/client/core/util/DefaultCharsetTest.java]: Are these lettered to assure ordering?

Comment (alexmoore) [inline src/test/java/com/basho/riak/client/core/util/DefaultCharsetTest.java]: They are, but I realized this smells more than I wanted. going to massage these tests a little more :/

**Comment by hazen** [comment]
Looks like we have some failing tests:

```
Failed tests:
  ITestRawIndexQuery.testBucketIndexHack:143 expected:<100> but was:<103>
  ITestSecondaryIndexQueryOp.testBucketIndexHack:540 null
  ITestDeleteOperation.testDeleteObjectTestType:46->testDeleteObject:73 expected:<{"value":"value"}> but was:<null>
```

**Comment by alexmoore** [comment]
How do you have your test-rig setup? Are we using the `java-client` branch of riak-client-tools?

**Comment by hazen** [comment]
This is from the buildbot tests

**Comment by lukebakken** [comment]
Attached is a zip from five test runs locally. Some were successful, some not.

[gh-593-test-runs.zip](https://github.com/basho/riak-java-client/files/137077/gh-593-test-runs.zip)

**Comment by alexmoore** [comment]
I wonder if oracle jdk would give different results.

Thanks,
Alex

**Comment by lukebakken** [comment]
OK I'll give it a try.

**Comment by lukebakken** [comment]
Unfortunately ... some of the same errors. I had to CTRL-C the last test, frozen on:

```
Running com.basho.riak.client.core.operations.itest.ITestListKeysOperation
```

[gh-593-oracle-test-runs.zip](https://github.com/basho/riak-java-client/files/137134/gh-593-oracle-test-runs.zip)

**Comment by alexmoore** [comment]
@javajolt @lukebakken Ok, this one should be ready now too. Fixes include all the security / buildbot fixes from PR #595.  This should include everything that you've noted in this review so far.

**Comment by lukebakken** [comment]
@alexmoore - is it worth making the changes to the tests at this point so the `java-client` branch in `riak-client-tools` can go away?

**Comment by alexmoore** [comment]
@lukebakken Now that we're using LevelDB as the default file system I think we can.  I'd like to not stack anymore stuff into this cascade of PRs :/  I can start on that after we get these merged - I _think_ I would need to change the bucket type names mostly, and then the other 80% unexpected work.

**Comment by lukebakken** [comment]
Great! Please to be shipping :+1:

**Pull request merged**